From d367b1ff473783201b696019be31925b9286c30f Mon Sep 17 00:00:00 2001
From: "Alex Xu (Hello71)" <alex_y_xu@yahoo.ca>
Date: Thu, 3 Mar 2022 22:06:47 -0500
Subject: [PATCH] remove -l flag to 7z a

7z a -l flag was added in ace277330b37 "[GSoC] Implement AddTo, Rename,
Copy, Move, Paste Actions and Jobs" without explanation. -l causes p7zip
to follow symlinks when adding files. this is inconsistent with other
ark plugins, unintuitive, and not supported by upstream 7zip. therefore,
remove it.
---
 autotests/plugins/cli7zplugin/cli7ztest.cpp | 5 -----
 plugins/cli7zplugin/cliplugin.cpp           | 3 +--
 2 files changed, 1 insertion(+), 7 deletions(-)

diff --git a/autotests/plugins/cli7zplugin/cli7ztest.cpp b/autotests/plugins/cli7zplugin/cli7ztest.cpp
index 3ce7d484..13a24832 100644
--- a/autotests/plugins/cli7zplugin/cli7ztest.cpp
+++ b/autotests/plugins/cli7zplugin/cli7ztest.cpp
@@ -281,7 +281,6 @@ void Cli7zTest::testAddArgs_data()
             << QString() << false << 5 << QStringLiteral("LZMA2") << 0UL
             << QStringList {
                    QStringLiteral("a"),
-                   QStringLiteral("-l"),
                    QStringLiteral("-mx=5"),
                    QStringLiteral("-m0=LZMA2"),
                    QStringLiteral("/tmp/foo.7z")
@@ -292,7 +291,6 @@ void Cli7zTest::testAddArgs_data()
             << QStringLiteral("1234") << false << 5 << QStringLiteral("LZMA2") << 0UL
             << QStringList {
                    QStringLiteral("a"),
-                   QStringLiteral("-l"),
                    QStringLiteral("-p1234"),
                    QStringLiteral("-mx=5"),
                    QStringLiteral("-m0=LZMA2"),
@@ -304,7 +302,6 @@ void Cli7zTest::testAddArgs_data()
             << QStringLiteral("1234") << true << 5 << QStringLiteral("LZMA2") << 0UL
             << QStringList {
                    QStringLiteral("a"),
-                   QStringLiteral("-l"),
                    QStringLiteral("-p1234"),
                    QStringLiteral("-mhe=on"),
                    QStringLiteral("-mx=5"),
@@ -317,7 +314,6 @@ void Cli7zTest::testAddArgs_data()
             << QString() << false << 5 << QStringLiteral("LZMA2") << 2500UL
             << QStringList {
                    QStringLiteral("a"),
-                   QStringLiteral("-l"),
                    QStringLiteral("-mx=5"),
                    QStringLiteral("-m0=LZMA2"),
                    QStringLiteral("-v2500k"),
@@ -329,7 +325,6 @@ void Cli7zTest::testAddArgs_data()
             << QString() << false << 5 << QStringLiteral("BZip2") << 0UL
             << QStringList {
                    QStringLiteral("a"),
-                   QStringLiteral("-l"),
                    QStringLiteral("-mx=5"),
                    QStringLiteral("-m0=BZip2"),
                    QStringLiteral("/tmp/foo.7z")
diff --git a/plugins/cli7zplugin/cliplugin.cpp b/plugins/cli7zplugin/cliplugin.cpp
index 1781117a..b136baee 100644
--- a/plugins/cli7zplugin/cliplugin.cpp
+++ b/plugins/cli7zplugin/cliplugin.cpp
@@ -50,8 +50,7 @@ void CliPlugin::setupCliProperties()
     m_cliProps->setProperty("captureProgress", false);
 
     m_cliProps->setProperty("addProgram", QStringLiteral("7z"));
-    m_cliProps->setProperty("addSwitch", QStringList{QStringLiteral("a"),
-                                                 QStringLiteral("-l")});
+    m_cliProps->setProperty("addSwitch", QStringList{QStringLiteral("a")});
 
     m_cliProps->setProperty("deleteProgram", QStringLiteral("7z"));
     m_cliProps->setProperty("deleteSwitch", QStringLiteral("d"));
-- 
2.36.1

